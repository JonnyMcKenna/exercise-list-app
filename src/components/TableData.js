import "../App.css";
import { useState } from "react";
import * as constants from "../constants";
import ImageComponent from "./ImageComponent";

const TableData = (props) => {
  const [viewInstructions, setViewInstructions] = useState(false);

  const {
    exercise: { name, male, female, transcript },
    gender,
  } = props;

  var extraContent = (
    <div
      className="extraContent"
      dangerouslySetInnerHTML={{ __html: transcript }}
    />
  );

  return (
    <tbody>
      <tr className="tableRow">
        <td className="firstColumn">
          <div className="exerciseName">{name}</div>

          <ImageComponent
            male={male}
            female={female}
            gender={gender}
            maleClassName={constants.MALE_IMAGE_MOBILE}
            femaleClassName={constants.FEMALE_IMAGE_MOBILE}
          />

          <button
            className="view-instructions-link"
            id="view-instructions-link-id"
            onClick={() => {
              setViewInstructions(!viewInstructions);
            }}
          >
            View Instructions
          </button>

          {viewInstructions && extraContent}
        </td>

        <td className="secondColumn">
          <ImageComponent
            male={male}
            female={female}
            gender={gender}
            maleClassName={constants.MALE_IMAGE}
            femaleClassName={constants.FEMALE_IMAGE}
          />
        </td>
      </tr>
    </tbody>
  );
};

export default TableData;
