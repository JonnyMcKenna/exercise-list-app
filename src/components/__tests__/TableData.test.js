import React from "react";
import TableData from "../TableData";
import { render } from "@testing-library/react";

describe("TableData", () => {
  it("should render the specific exercise", () => {
    const exercise = [
      {
        id: "123",
        name: "name",
        transcript: "transcript",
        male: { image: "image" },
        female: { image: "image" },
      },
    ];
    const appElement = render(
      <TableData exercise={exercise} gender="male" />
    ).baseElement;
    expect(appElement).toMatchSnapshot();
    expect(appElement).toHaveTextContent("View Instructions");
  });
});
