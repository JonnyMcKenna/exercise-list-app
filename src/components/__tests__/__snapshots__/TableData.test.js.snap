// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableData should render the specific exercise 1`] = `
<body>
  <div>
    <tbody>
      <tr
        class="tableRow"
      >
        <td
          class="firstColumn"
        >
          <div
            class="exerciseName"
          />
          <div>
            <img
              alt="Male"
              class="maleImageMobile"
              src=""
            />
          </div>
          <button
            class="view-instructions-link"
            id="view-instructions-link-id"
          >
            View Instructions
          </button>
        </td>
        <td
          class="secondColumn"
        >
          <div>
            <img
              alt="Male"
              class="maleImage"
              src=""
            />
          </div>
        </td>
      </tr>
    </tbody>
  </div>
</body>
`;
